src/ray/core_worker/core_worker.h

```diff
@@ -104,28 +106,47 @@ class CoreWorker {
   ///
   /// \param[in] object_id The object ID to increase the reference count for.
   void AddObjectIDReference(const ObjectID &object_id) {
-    reference_counter_->AddReference(object_id);
+    reference_counter_->AddLocalReference(object_id);
   }
 
   /// Decrease the reference count for this object ID.
   ///
   /// \param[in] object_id The object ID to decrease the reference count for.
   void RemoveObjectIDReference(const ObjectID &object_id) {
     std::vector<ObjectID> deleted;
-    reference_counter_->RemoveReference(object_id, &deleted);
+    reference_counter_->RemoveLocalReference(object_id, &deleted);
     if (ref_counting_enabled_) {
       memory_store_->Delete(deleted);
     }
   }
 
-  /// Promote an object to plasma. If it already exists locally, it will be
-  /// put into the plasma store. If it doesn't yet exist, it will be spilled to
-  /// plasma once available.
+  /// Serialize an ObjectID. This will promote the object to plasma. If it
+  /// already exists locally, it will be put into the plasma store. If it
+  /// doesn't yet exist, it will be spilled to plasma once available.
   ///
   /// Postcondition: Get(object_id.WithPlasmaTransportType()) is valid.
   ///
-  /// \param[in] object_id The object ID to promote to plasma.
-  void PromoteObjectToPlasma(const ObjectID &object_id);
+  /// \param[in] object_id The object ID to serialize.
+  /// \param[out] owner_id The ID of the object's owner. This should be
+  /// appended to the serialized object ID.
+  /// \param[out] owner_address The address of the object's owner. This should
+  /// be appended to the serialized object ID.
+  /// \return Whether we had the object's owner or not. If we don't have the
+  /// object's owner, then calling ray.get on the deserialized object ID may
+  /// hang if the object is evicted.
+  bool SerializeObjectId(const ObjectID &object_id, TaskID *owner_id,
+                         rpc::Address *owner_address);
```

Comment:
Should we add a proto definition to put together `object_id`, `owner_id`, and `owner_address`, probably called `BorrowedObjectID` ?

Reply:
Hmm probably makes sense, but I don't think it's necessary yet. It's nice to see exactly what information is needed, and we're not planning on adding any additional fields.